src/styles/transitions/BaseTransition.tsx

```diff
@@ -16,28 +16,43 @@ export const BaseTransitionPropTypes = {
   ]),
 };
 
-const BaseTransition = ({ children, in: inProp, ...props }) => {
-  const delayTimeout = 500;
-  const delayedInProp = useDelayedChange(inProp, delayTimeout);
+interface IBaseTransitionProps
```

Comment:
Refactored this to TypeScript since it was pretty simple and it was giving me 30000 compilation errors after adding the `delayTimeout` prop